### DISCUSSION TITLE
macOS Debug message log

### DISCUSSION THREAD

**Discussion by GaryElshaw** [transmission/transmission#4017] [Q&A]
### What is the issue?

It's arisen before, but the Debug log is currently absolutely massive when running for a few hours. A couple of thoughts arise. Can or should it be pruned at a particular time interval? What are the potential memory implications for this much data being retained?

![SCR-20221024-oxp](https://user-images.githubusercontent.com/69029666/197452046-664921f7-ad2f-44c5-8641-8131710b773c.png)

### Which application of Transmission?

macOS app

### Which version of Transmission?

4.0.0-beta.1.dev (cee339e10d)

**Comment by ckerr** [comment]
Wellll... hm.

There's currently code to limit repetitive _warnings and errors_ (added in https://github.com/transmission/transmission/pull/2756) since those are always enabled, but it doesn't apply to opt-in messages like debug and trace. 

IMO the use case for debug / trace is that you want to drink from the firehose because you're trying to diagnose a problem; it's not something intended for daily use. It probably does consume a lot of memory but it's not going to affect the average user since they don't opt in for debug messages.

**Comment by GaryElshaw** [comment]
Is there an attribute i can change in the json to disable trace and/or debug, or does this have to be done during the build process?
